### ISSUE TITLE
Variable concatenation adds a space in between

### ISSUE THREAD

**Issue by ausi** [sass/sass#1704]
Version: 3.4.13
Syntax: SCSS and intended syntax

If you concatenate two variables or a literal with a variable one space character gets added between the two:

``` sass
$input: literal;
.result {
  output: literal$input;
  output: $input$input;
}
```

``` css
.result {
  output: literal literal;
  output: literal literal; }
```

This only happens for these two combinations, all other combinations of `literal`, `$input`, `#{$input}`, `#{literal}`, `#{"literal"}` get concatenated without spaces. You can find a complete test case in sass/sass-spec#325: [input.scss](https://github.com/sass/sass-spec/blob/e0a26d0ddad5b2e586766076dcd94337fcbdf2d5/spec/parser/interpolate/01_literal/07_variable_concatenation/input.scss).

**Comment by nex3** [comment]
This is intended. When two distinct values are next to one another, whether or not there's whitespace between them in the source, they're parsed as elements of a list. `literal$input` is the same as `literal $input` is the same as `(literal $input)`; the parallel is true for `$input$input`.

Interpolation works differently. It's never considered distinct from a neighboring value; its presence makes Sass try to parse everything touching it as an unquoted string, with the interpolation otherwise working as it would in a quoted string. So `literal#{$input}` means the same thing as `unquote("literal#{$input}")`.

**Comment by ausi** [comment]
Thanks for the explanation and sorry for the inconvenience.

**Comment by MohamedLamineAllal** [comment]
`node-sass` behavior was different than that. I worked to port SnippetStore (I'm not the author) to a newer version of electron. To support apple silicon (arm64). And part of the porting, I had to remove `node-sass`. And replace it with `sass`. And what was working stopped working.

https://github.com/ZeroX-DG/SnippetStore/blob/f2788f24785d83d688afe37f2328830fe346ec79/browser/render/lib/styles/utility.sass#L43

```sass
@for $i from 1 through 200
  .#{m}-#{$i}
    margin: $i#{'px'}!important
  .#{m-l}-#{$i}
    margin-left: $i#{'px'} !important
  .#{m-r}-#{$i}
    margin-right: $i#{'px'} !important
  .#{m-t}-#{$i}
    margin-top: $i#{'px'} !important
  .#{m-b}-#{$i}
    margin-bottom: $i#{'px'} !important
  ```
  
  `m-l-10` ==> give a CSS style like this `margin-left: 10 px !important`. the space between 10 and px breaks the utility.
  
  It was working just well with `node-sass`. So it is a breaking change.
  
  I can fix the utility. By using the interpolation differently. luckily there aren't many of them.

Ex: 

```sass
.#{m-l}-#{$i}
    margin-left: #{$i}px !important
```

using the interpolation on the variable.
